### ISSUE TITLE
Lighgbm model performace on old data after retraining on new data using 'init_model'

### ISSUE THREAD

**Issue by ninenerd** [microsoft/LightGBM#3781]
I have divided my data into 2 parts. 

Train model on **part1** data and save the file. 
Reload the saved model and resume training using init_model and on new data **part2**.  

Now that I am checking new model results on part1 data and accuracy has decreased a bit. 
Is it expected to lose accuracy on old data after retraining.  

Any changes that I can do here ?

Edit : Also when I am fitting new data on model using init_model, do i have to provide previous model parameters as well ? 

`lgbm_model = LGBMRegressor()` # Should i provide previous models params such as max_depth,min_child_samples, subsample ?
` lgbm_model.fit(X_new,y_new,init_model=model.txt`)

**Comment by StrikerRUS** [comment]
> Is it expected to lose accuracy on old data after retraining.

It highly depends on your data.  You cannot expect the continued training will always improve performance on a fixed validation set.

Using `init_model` only means that your training on **new** data will start from `init_model`'s predictions.
So, 
> Also when I am fitting new data on model using init_model, do i have to provide previous model parameters as well ?

you should provide **any parameters** that you think will help to describe your **new** data best. It is not necessary that best params = old ones.

**Comment by ninenerd** [comment]
```
lgbm_gridF = RandomizedSearchCV(LGBMRegressor(random_state=random_seed), lgbm_param_dict, cv = 3, verbose = 1, scoring = 'neg_root_mean_squared_error', random_state = 28, n_jobs = -1, refit=True, n_iter=200)
    lgbm_bestF = lgbm_gridF.fit(X,y,init_model=model.txt)

    best_lgbm = lgbm_bestF.best_estimator_
    best_lgbm.fit(X,y,init_model=model.txt)
```
To counter the param issue, I have done HPT for new model as well. 
Is this correct ?

**Comment by StrikerRUS** [comment]
Yep.
